File: leetcode-tasks/2894-divisible-and-non-divisible-sums-difference.ts
// Accepted
// Runtime 55 ms Beats 89.92% of users with TypeScript
// Memory 51.79 MB Beats 88.24% of users with TypeScript

const differenceOfSums = (n: number, m: number): number => {
  let nums1 = 0
  let nums2 = 0

  for (let i = 1; i <= n; i++) {
    if (i % m !== 0) nums1 += i
  }
  for (let i = m; i <= n; i += m) {
    nums2 += i
  }

  return nums1 - nums2
}

File: tasks/2894-divisible-and-non-divisible-sums-difference.go
// Accepted 737 / 737 testcases passed Sergey Pomortsev submitted at Jul 05, 2026 13:04
// Runtime 0 ms Beats 100.00% Memory 4.16 MB Beats 36.84%

func differenceOfSums(n int, m int) int {
  var div int
  var nonDiv int

  for i := 1; i <= n; i++ {
    if i % m != 0 {
      div += i
    } else {
      nonDiv += i
    }
  }

  return div - nonDiv
}